### ISSUE TITLE
If `onChange` passed to `Field` component returns truthy value, then component don't get field value updated

### ISSUE THREAD

**Issue by finom** [erikras/redux-form#4482]
### Are you submitting a **bug report** or a **feature request**?
**bug report**

### What is the current behavior?
If `onChange` passed to `Field` component returns truthy value, then component don't get field value updated.

A component from the sandbox below: 

```
<Field
      name="name"
      component={({ input: { value, onChange } }) => (
        <div onClick={() => onChange(Math.random())}>click me {value}</div>
      )}
      placeholder="Name"
      onChange={() => true}
    />
```
If you'd click on "click me" text you'll **not** get a new random value. But if you'd replace `onChange={() => true}` by `onChange={() => false}` you'll get it functioning.

At my specific case onChange returned `setTimeout` value to get redux sagas notified after store is updated (`setTimeout` as you know returns timeout ID). It was really challenging to understand what's going on after I decided to finally update react-redux and redux-form. 

### What is the expected behavior?
Component updates shouldn't depend on what onChange returns.

### Sandbox Link
All core dependencies got updated https://codesandbox.io/s/reduxform-template-87mjk

### What's your environment?
No matter. The bug is reproduce-able on any environment.

### Other information
Didn't find any information about this bug.

**Comment by Senderek** [comment]
I'm also a victim of this bug, moreover the bug doesn't exist up to at least version 7.4.2. Is there any chance for a fix?

**Comment by anupalhub** [comment]
@finom Any solution you got?

**Comment by finom** [comment]
@anupalhub nope, just make all functions not return anything: `onChange={() => { onChange() }}`
